File: day1/day1.jl
f = open("input1.txt", "r")
data = []
for l in eachline(f) 
    append!(data, parse(Int64, l))
end

function answer(data) 
    for i = 1:length(data)
        for j = i + 1:length(data) 
            if data[i] + data[j] == 2020 
                println(data[i], " ", data[j], " ", data[i] * data[j])
                return
            end
        end
    end
end

function answer2(data)
    for i = 1:length(data)
        for j = i + 1:length(data) 
            for k = i + 2:length(data)
                if data[i] + data[j] + data[k] == 2020 
                    println(data[i], " ", data[j], " ", data[k], " ", data[i] * data[j] * data[k])
                    return
                end
            end
        end
    end
end

answer(data)
answer2(data)

File: day1/day1.py
filename = "input1.txt"
data = []
with open(filename) as f:
    for line in f.readlines():
        data.append(int(line))
    f.close()

print(len(data))
done = 0
for i in range(0, len(data)):
    for j in range(i+1, len(data)):
        for k in range(i+2, len(data)):
            if data[i]+data[j]+data[k] == 2020:
                print(i, j, k, data[i], data[j],
                      data[k], data[i]*data[j]*data[k])
                done = 1
                break  # k loop
        if done == 1:
            break  # j loop
    if done == 1:
        break  # i loop
